### ISSUE TITLE
[BUG] PR Labeler doesn't work right

### ISSUE THREAD

**Issue by wwahammy** [houdiniproject/houdini#276]
The labeler tool for Github actions doesn't properly label PRs. It seems like it's acting like one of labels provided in [labeler.yml](https://github.com/houdiniproject/houdini/blob/chore/fix-labeler/.github/labeler.yml) is missing but I don't see where it would be.

## Steps to Reproduce

1. Submit a PR that modifies some set of files that should trigger a label to be added.
2. The action fails and no labels are added. As an example, see: https://github.com/houdiniproject/houdini/pull/274 and click "View details" where it mentions the commit was merged.

**Comment by fellipepcs** [comment]
Hello wwahammy!

Are you still wanting this functionality implemented?


Dafne and I are working on this issue and have opened a Pull request in WIP https://github.com/houdiniproject/houdini/pull/1488
But we have some doubts.
we create a label directly in the dependabot.yml file.

We also created a test using the labeler, but we don't know how we are going to test it. Can you help us with this?

**Comment by wwahammy** [comment]
@fellipepcs We do want this implemented! Testing of Github actions and related features are a huge pain. My recommendations for testing that are as follows:

The best way to approach this, like many complex problems in software engineeering, is to start from the simplest version of the problem, solve it and then add complexity until you've solved the actual problem you want to solve.

Simplify the problem by creating your own personal repo with the smallest subset of features you need. Create a repo that has just the minimal stuff needed to use [the PR labeler]https://github.com/actions/labeler). Then, once you think you have it set up, create PRs on your personal repo to test the various conditions. For example. let's say you have it set up to add the Ruby label to PRs where any file matching `**/*.rb` files is modified. Create a new PR in your personal repo where you add or modify a file with a .rb prefix in the root directory. (I think can use something like "add file" in the Github UI to create a new branch and PR to your main branch.) Once the PR is created, see if the labeling works. If it doesn't, close the PR and try to figure out why it didn't work and then try again. If it does, close the PR and create a new PR with a file with a .rb prefix in subdirectory. Check if the labeling work and if it does try an .rb prefix in a subdirectory of a subdirectory. If you're testing `**/*.rb`, a subdirectory of a subdirectory is far enough to go.

Try every label and every condition (or a reasonable subset) and prove that they work.

Also, test that when PRs are changed modified, the labeler action runs ( you should be able to see that in your actions tab on the repo)

Once you have the labels working as expected, update your PR with the changes you created related labeling in your personal repo. Then link to the PRs in your test repo to and explain what conditions you were testing for each PR

Does that make sense?

**Comment by DGrothe-PhD** [comment]
It looks like the labeler doesn't work because of missing `any`'s.
So I suggest the following:
1. .github/labeler.yml: change every line to the syntax
```
ruby:
- any: ['src/**/*.rb']
anotherlabel:
- any: ['some/path/*.pdf', 'yet/another/path/**/*.txt']
```
2. Be sure that every label mentioned in labeler.yml really exists. (Otherwise, just create it, and set up a color for it)
3. If it is still not working, rename [label.yml](https://github.com/houdiniproject/houdini/blob/chore/fix-labeler/.github/workflows/label.yml) to *labeler.yml*

See also https://github.com/actions/labeler for reference.

**Comment by DGrothe-PhD** [comment]
Okay, I have set up this https://github.com/houdiniproject/houdini/pull/1573 pull request. To avoid any merge conflicts, I did not directly submit a PR on `main` as `main` hasn't got labeler workflow; however, the branch https://github.com/houdiniproject/houdini/tree/chore/fix-labeler is plenty of commits behind main.
Consider cherry-picking these changes on https://github.com/houdiniproject/houdini/tree/chore/fix-labeler back to main or synching that one with `main` first.

**Comment by DGrothe-PhD** [comment]
Closed #1573 and opened #1574 which worked well on my test branch.

**Comment by wwahammy** [comment]
Hopefully closed by https://github.com/houdiniproject/houdini/pull/1574
